api/test/integration/test_webhook_endpoints.tavern.yaml

```diff
@@ -0,0 +1,174 @@
+    request: &webhook_request
+      verify: false
+      url: "{protocol:s}://{host:s}:{port:d}/events"
+      method: POST
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      json:
+        events:
+          - foo event
+    response:
+      status_code: 200
+      json:
+        data:
+          affected_items:
+            - foo event
+          total_affected_items: 1
+          total_failed_items: 0
+          failed_items: []
+        message: All events were forwarded to analisysd
+        error: 0
+
+  - name: POST /events with multiple elements
+    request:
+      verify: false
+      <<: *webhook_request
+      json:
+        events:
+          - foo event
+          - bar event
+          - gaz event
+    response:
+      status_code: 200
+      json:
+        data:
+          affected_items:
+            - foo event
+            - bar event
+            - gaz event
+          total_affected_items: 3
+          total_failed_items: 0
+          failed_items: []
+        message: All events were forwarded to analisysd
+        error: 0
+
+
+---
+test_name: POST /events with invalid formats
+
+stages:
+  - name: POST /events with invalid event type
+    request:
+      verify: false
+      <<: *webhook_request
+      json:
+        events:
+          - "foo": 1
+    response:
+      status_code: 400
+      json:
+        detail: "{{'foo': 1}} is not of type 'string' - 'events.0'"
+        title: Bad Request
+
+  - name: POST /events with invalid body
+    request:
+      verify: false
+      <<: *webhook_request
+      json:
+        - foo
+        - bar
+    response:
+      status_code: 400
+      json:
+        detail: "['foo', 'bar'] is not of type 'object'"
+        title: Bad Request
+
+---
+
+test_name: POST /events with more than 100 events
+
+marks:
+  - usefixtures:
+      - big_events_payload
+
+stages:
+  - name: POST /events with invalid size
+    request:
+      verify: false
+      <<: *webhook_request
+      json:
+        events: !force_format_include "{big_events_payload}"
+    response:
+      status_code: 400
+      json:
+        detail: The size of the events bulk is exceeding the limit
+        title: Events bulk size exceeded
+
+---
+
+test_name: POST /events with big events
+
+marks:
+  - usefixtures:
+      - max_size_event
+      - large_event
+
+stages:
+  - name: POST /events with larger event (65536B)
+    request:
+      verify: false
+      <<: *webhook_request
+      json:
+        events:
+          - "{large_event}"
+    response:
+      status_code: 200
+      json:
+        data:
+          affected_items: []
+          total_affected_items: 0
+          total_failed_items: 1
+          failed_items:
+            - error:
+                code: 1012
+                message: "Invalid message to queue: The event is too large to be sent to analysisd (maximum is 65535B)"
+              id:
+                - "{large_event}"
+        message: No events were forwarded to analisysd
```

Comment:
Error code should also be checked (`"error": 2`).

Reply:
Here I'm getting `"error": 1` and in the stage above `"error": 2`, is this ok? Or is a bad behavior?

The implementation is just returning the default affected item's results.

https://github.com/wazuh/wazuh/blob/0a7439376a1c4082fe4c48f158f83e785e8738a1/framework/wazuh/event.py#L24-L39

Reply:
Yes, sorry, in this case it would be `"error": 1`. The error code is added when rendering AffectedItemsWazuhResult. The code depends on the amount of failed items. 
- **No failed items**: error code 0
- **All failed items:** error code 1
- **Partial, there are affected and failed items:** error code 2